authtools/backends.py

```diff
@@ -0,0 +1,19 @@
+from django.contrib.auth.backends import ModelBackend
+
+
+class CaseInsensitiveEmailBackend(ModelBackend):
+    """
+    This authentication backend assumes that usernames are email addresses and simply lowercases
+    a username before an attempt is made to authenticate said username using Django's ModelBackend.
+
+    Example usage:
+        # In settings.py
+        AUTHENTICATION_BACKENDS = ('authtools.backends.CaseInsensitiveEmailBackend',)
+    """
+    def authenticate(self, username=None, password=None):
```

Comment:
I have an app that requires me to pass in the request to the authenticate calls.